lib/modules/dosomething/dosomething_home/dosomething_home.module

```diff
@@ -82,7 +79,17 @@ function dosomething_home_preprocess_node(&$vars) {
 
       $tiles['status'] = TRUE;
       $tiles['image'] = '<img src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs=" data-src="' . $tiles['image_url'] . '">';
-      $tiles['link'] = drupal_get_path_alias('node/' . $nid) . $source;
+
+      $url_options = array(
+        'language' => $lang_code,
+        'query' => array(
+          'source' => 'node/' . $vars['nid'],
+        ),
+      );
+      if ($country = strtolower(dosomething_global_convert_language_to_country($lang_code))) {
+        $url_options['prefix'] = $country . '/';
+      }
+      $tiles['link'] = url('node/' . $nid, $url_options);
```

Comment:
The url() function automatically will return an alias, and giving it the 'language' param will give you the appropriately translated version of the url, like http://dev.dosomething.org:8888/br/campanhas/gtfo-pt-br?source=node/1141